packages/application-config/src/load-config.ts

```diff
@@ -1,32 +1,51 @@
 import type { JSONSchemaForCustomApplicationConfigurationFiles } from './schema';
+import type { LoaderSync } from 'cosmiconfig';
 
 import { cosmiconfigSync, defaultLoaders } from 'cosmiconfig';
+import get from 'lodash/get';
+
+const loadJsModule: LoaderSync = (filePath) => {
+  // Load JS modules using Babel, as we need to load
+  // the config synchronously with `require`, no `await import`.
+  require('@babel/register')({
+    babelrc: false,
+    extensions: ['.js', '.cjs', '.mjs', '.ts'],
+    presets: ['@commercetools-frontend/babel-preset-mc-app'],
+  });
+  const result = require(filePath);
+  return get(result, 'default', result);
+};
```

Comment:
This is the custom loader, as cosmiconfig does not handle `.mjs` and `.ts`. For simplicity and consistency, I decided to use the same loader also for `.js` and `.cjs` files.

Note that we use our own babel preset, so it can handle both JS and TS files.

And we use babel because we want to load the file synchronously, otherwise if we use `import` we need to make things `async`, which is not possible for the app config.